### PR TITLE
Add initial implementation

### LINKED ISSUE #2: Initial implementation

### PR DESCRIPTION
**Pull request by KevinGrandon**
Initial implementation of an apollo renderer.

Fixes #2

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      } else {
+        return next();
+      }
+    };
+
+    this.plugin(() => preRenderPlugin);
+  }
+}
+
+export {ProviderPlugin, ProvidedHOC, Provider};
```
Comment (KevinGrandon) [inline src/index.js]: Bunch of duplicate code here due to exporting these. Not sure if I should rip this out or not. Thoughts?

Comment (ganemone) [reply src/index.js]: Yea we could potentially pull this out, however I'm not totally sure how I feel about this abstraction in the first place.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import ProviderPlugin from './plugin';
+import ProvidedHOC from './hoc';
+import Provider from './provider';
+
+export default class App extends CoreApp {
+  constructor(root, getClient) {
+    super(root, el => {
+      return prepare(el).then(() => {
+        return __NODE__ ? serverRender(el) : clientRender(el);
```
Comment (ganemone) [inline src/index.js]: Might want to consider supporting passing through a render function here to support testing via enzyme

File: package.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    "build-test": "rm -rf dist-tests && cup build-tests",
+    "just-test": "node_modules/.bin/unitest --browser=dist-tests/browser.js --node=dist-tests/node.js",
+    "test": "npm run build-test && npm run just-test",
+    "prepublish": "npm run transpile"
+  },
+  "peerDependencies": {
+    "fusion-core": "^0.1.4",
+    "fusion-react-async": "^0.1.4",
+    "react": "14.x - 16.x",
+    "react-dom": "14.x - 16.x"
```
Comment (ganemone) [inline package.json]: i think `react-apollo` be a `peerDependency` here

File: src/index.js (hunk 2 above)
Comment (KevinGrandon) [reply src/index.js]: I'll work on adding an example enzyme test, but I might make that a follow-up.

File: src/index.js (hunk 1 above)
Comment (KevinGrandon) [reply src/index.js]: I've gone ahead and pulled these out to make this PR more simple.

**Pull request merged**